tests/v2/validation/snapshot/snapshot_restore_k8s_upgrade_test.go

```diff
@@ -0,0 +1,108 @@
+//go:build (validation || extended) && !infra.any && !infra.aks && !infra.eks && !infra.gke && !infra.rke2k3s && !cluster.any && !cluster.custom && !cluster.nodedriver && !sanity && !stress
+
+package snapshot
+
+import (
+	"testing"
+
+	"github.com/rancher/rancher/tests/framework/clients/rancher"
+	"github.com/rancher/rancher/tests/framework/extensions/etcdsnapshot"
+
+	"github.com/rancher/rancher/tests/framework/pkg/config"
+	"github.com/rancher/rancher/tests/framework/pkg/session"
+
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+)
+
+type SnapshotRestoreK8sUpgradeTestSuite struct {
+	suite.Suite
+	session        *session.Session
+	client         *rancher.Client
+	clustersConfig *etcdsnapshot.Config
+}
+
+func (s *SnapshotRestoreK8sUpgradeTestSuite) TearDownSuite() {
+	s.session.Cleanup()
+}
+
+func (s *SnapshotRestoreK8sUpgradeTestSuite) SetupSuite() {
+	testSession := session.NewSession()
+	s.session = testSession
+
+	s.clustersConfig = new(etcdsnapshot.Config)
+	config.LoadConfig(etcdsnapshot.ConfigurationFileKey, s.clustersConfig)
+
+	client, err := rancher.NewClient("", testSession)
+	require.NoError(s.T(), err)
+
+	s.client = client
+}
+
+func (s *SnapshotRestoreK8sUpgradeTestSuite) TestRKE1SnapshotRestoreK8sUpgrade() {
```

Comment:
Markus what happens when you run the whole test suite and the cluster is a k3s cluster? Will it run this test? We need to be able to run the whole test suite by default.

Reply:
Fair point. I was trying to not have to create a separate RKE1 test file for each of these tests. I can see if maybe I can do some kind of check for RKE2/K3S and skip it. If not, I don't see how we get around not making 3 new test files for RKE1.

Reply:
It's either this test can be ran on all cluster types, or we need separate them and tag them accordingly. We need to be able to run the whole test suite in recurring runs. We have tags to distinguish what tests to run on what clusters. We can't design our suites like this. We have to keep in mind that the whole test suite needs to be ran. Don't you already have logic to distinguish? If we do, why is there a need to write it like this?

Reply:
You're right actually, because of the new `IsRKE1` boolean I added, we should be good to to remove these functions. Let me do that, run a quick test and make sure. But I believe that will work.

Reply:
Could you run the full suite on all cluster types. If that is possible you are the man!